src/com/company/Game.java

```diff
@@ -0,0 +1,186 @@
+                onStrike = battingTeam.getMember().get(nextBatsman);
+                nextBatsman++;
+            }
+
+        }
+        else if(outcome.equals("0") || outcome.equals("2") || outcome.equals("4") || outcome.equals("6")) {
+            battingTeam.setScore(battingTeam.getScore() + Integer.parseInt(outcome));
+            onStrike.setRunsScored(onStrike.getRunsScored() + Integer.parseInt(outcome) );
+            onStrike.setBallsFaced(onStrike.getBallsFaced() + 1);
+            bowler.setRunsGiven(bowler.getRunsGiven() + Integer.parseInt(outcome));
+
+            System.out.println("Score on this ball: " + outcome + "run/s");
+        }
+        else {
+            battingTeam.setScore(battingTeam.getScore() + Integer.parseInt(outcome));
+            onStrike.setRunsScored(onStrike.getRunsScored() + Integer.parseInt(outcome) );
+            onStrike.setBallsFaced(onStrike.getBallsFaced() + 1);
+            bowler.setRunsGiven(bowler.getRunsGiven() + Integer.parseInt(outcome));
+
+            System.out.println("Score on this ball: " + outcome + "run/s");
+            Player temp =  onStrike;
+            onStrike = nonStrike;
+            nonStrike = temp;
+        }
+        bowler.setBallsDelivered(bowler.getBallsDelivered() + 1);
+        return nextBatsman;
+    }
+
+    //for an inning, made a method which is responsible for conducting overs of an inning, in it bowlers are changed, ballOutcome method is called repeatedly (also nextPlayer variable is passed in it so that which player will come to bat next can be tracked) . It is also responsible for change of strike after every over, and also prints score after every over and detailed info at the end of the inning
+    public void inning(boolean secondInning){
+        onStrike = battingTeam.getMember().get(0);
+        nonStrike = battingTeam.getMember().get(1);
+        int nextBatsman = 2,bowlAt = 0,i,j;
+        for( i=0;i<overs && battingTeam.getNotOut()>1;i++)
+        {
+            bowler = bowlingTeam.getMember().get(battingTeam.getPlayers() - 1 - bowlAt);
+            for( j=0;j<6 && battingTeam.getNotOut()>1;j++) {
+                nextBatsman = ballOutcome(nextBatsman);
+                if(secondInning && (battingTeam.getScore()>bowlingTeam.getScore()) )
+                {
+                    break;
+                }
+
+            }
+
+            System.out.println("****");
+            System.out.println("Score after "+(i+1)+"."+(bowler.getBallsDelivered())%6+" over(s) is "+ battingTeam.getScore() );
+            System.out.println(onStrike.getName() + " : " + onStrike.getRunsScored()+"("+ onStrike.getBallsFaced()+")");
+            System.out.println(nonStrike.getName() + " : " + nonStrike.getRunsScored()+"("+nonStrike.getBallsFaced()+")");
+            System.out.println("****");
+            if(secondInning && (battingTeam.getScore()>bowlingTeam.getScore()) )
+            {
+                break;
+            }
+            Player temp =  onStrike;
```

Comment:
swapping of onStrike and nonStrike batsman is happening at 2 places in code. extract it in a method and use it all places.

Reply:
Done